components/FlexibleHeader/tests/unit/FlexibleHeaderTraitColletionDidChangeTests.m

```diff
@@ -0,0 +1,117 @@
+  XCTestExpectation *expectation =
+      [self expectationWithDescription:@"Called traitCollectionDidChange"];
+  __block UITraitCollection *passedTraitCollection;
+  __block MDCFlexibleHeaderView *passedFlexibleHeader;
+  flexibleHeader.traitCollectionDidChangeBlock =
+      ^(MDCFlexibleHeaderView *_Nonnull flexibleHeaderView,
+        UITraitCollection *_Nullable previousTraitCollection) {
+        passedTraitCollection = previousTraitCollection;
+        passedFlexibleHeader = flexibleHeaderView;
+        [expectation fulfill];
+      };
+
+  // When
+  UITraitCollection *testCollection = [UITraitCollection traitCollectionWithDisplayScale:77];
+  [flexibleHeader traitCollectionDidChange:testCollection];
+
+  // Then
+  [self waitForExpectations:@[ expectation ] timeout:1];
+  XCTAssertEqual(passedTraitCollection, testCollection);
+  XCTAssertEqual(passedFlexibleHeader, flexibleHeader);
+}
+
+- (void)
+    testTraitCollectionDidChangeBlockCalledWhenTraitCollectionChangesForFlexibleHeaderViewController {
+  // Given
+  MDCFlexibleHeaderViewController *flexibleHeaderController =
+      [[MDCFlexibleHeaderViewController alloc] init];
+  XCTestExpectation *expectation =
+      [self expectationWithDescription:@"Called traitCollectionDidChange"];
+  flexibleHeaderController.traitCollectionDidChangeBlock =
+      ^(MDCFlexibleHeaderViewController *_Nonnull flexibleHeaderViewController,
+        UITraitCollection *_Nullable previousTraitCollection) {
+        [expectation fulfill];
+        flexibleHeaderViewController.traitCollectionDidChangeBlock = nil;
+      };
+
+  // When
+  [flexibleHeaderController traitCollectionDidChange:nil];
+
+  // Then
+  [self waitForExpectations:@[ expectation ] timeout:1];
+}
+
+- (void)
+    testTraitCollectionDidChangeBlockCalledWithExpectedParametersForFlexibleHeaderViewController {
+  // Given
+  MDCFlexibleHeaderViewController *flexibleHeaderController =
+      [[MDCFlexibleHeaderViewController alloc] init];
+  XCTestExpectation *expectation =
+      [self expectationWithDescription:@"Called traitCollectionDidChange"];
+  __block UITraitCollection *passedTraitCollection;
+  __block MDCFlexibleHeaderViewController *passedFlexibleHeader;
+  flexibleHeaderController.traitCollectionDidChangeBlock =
+      ^(MDCFlexibleHeaderViewController *_Nonnull flexibleHeaderViewController,
+        UITraitCollection *_Nullable previousTraitCollection) {
+        passedTraitCollection = previousTraitCollection;
+        passedFlexibleHeader = flexibleHeaderViewController;
+        [expectation fulfill];
+        flexibleHeaderViewController.traitCollectionDidChangeBlock = nil;
+      };
+
+  // When
+  UITraitCollection *testCollection = [UITraitCollection traitCollectionWithDisplayScale:77];
+  [flexibleHeaderController traitCollectionDidChange:testCollection];
+
+  // Then
+  [self waitForExpectations:@[ expectation ] timeout:1];
+  XCTAssertEqual(passedTraitCollection, testCollection);
```

Comment:
does this pass? passedTraitCollection doesn't get set until the block gets called. that happens after this line runs...

If you want to assert things are passed in correctly then I would put this assert inside the block

Reply:
I believe the assertions wait for the expectation to fulfill before they are ran.